src/libgit2/odb_mempack.c

```diff
@@ -132,6 +132,35 @@ static int git_mempack__dump(
 	return err;
 }
 
+int git_mempack_write_thin_pack(git_odb_backend *backend, git_packbuilder *pb)
+{
+	struct memory_packer_db *db = (struct memory_packer_db *)backend;
+	const git_oid *oid;
+	size_t iter = 0;
+	int err = -1;
```

Comment:
It looks like we don't depend on the initial condition? A minor nit, but

```suggestion
	int err;
```